pkg/cmd/server/api/v1/types.go

```diff
@@ -396,6 +396,8 @@ type MasterClients struct {
 
 // DNSConfig holds the necessary configuration options for DNS
 type DNSConfig struct {
+	// ClusterDomain is the authoritative domain of the DNS server.
+	ClusterDomain string `json:"clusterDomain"`
```

Comment:
in the `conversions.go` file in this package, set this to "cluster.local" if empty in this function:
```
		func(obj *DNSConfig) {
			if len(obj.BindNetwork) == 0 {
				obj.BindNetwork = "tcp4"
			}
		},
```